src/power_grid_model_io/utils/json.py

```diff
@@ -0,0 +1,90 @@
+# SPDX-FileCopyrightText: 2022 Contributors to the Power Grid Model IO project <dynamic.grid.calculation@alliander.com>
+#
+# SPDX-License-Identifier: MPL-2.0
+"""
+Custom json encoding functionalities
+"""
+
+import json
+from typing import IO, Any
+
+import numpy as np
+
+
+class JsonEncoder(json.JSONEncoder):
+    """
+    Custom JSON encoder for numpy types
+    """
+
+    def default(self, o):
+        if isinstance(o, np.integer):
+            return int(o)
+        if isinstance(o, np.floating):
+            return float(o)
+        if isinstance(o, np.ndarray):
+            return o.tolist()
+        return super().default(o)
+
+
+def compact_json_dump(data: Any, io_stream: IO[str], indent: int, max_level: int, level: int = 0):
```

Comment:
This is just a copy from power-grid-model, with the addition of `cls=JsonEncoder` in the `json.dump` calls.